### DISCUSSION TITLE
[form] Input text field inside radio button group

### DISCUSSION THREAD

**Discussion by ahetawal-p** [mantinedev/mantine#2188] [Q&A] [answered]
Trying to use marine/form with a multiple choice radio button group, with the last option as other (with input text).
How can I achieve this along with validation on the other option as well (textinput not empty when selected) for a required filed ?

**Comment by rtivital** [comment, marked as answer]
You can validate input based on other form values – https://mantine.dev/form/validation/#validation-based-on-other-form-values

**Reply by ahetawal-p** [reply]
Thanks, for the first part how can I show the other option with input text inside radio group. ?

**Reply by rtivital** [reply]
`form.values.radioGroup === 'option with input' && <TextInput />`

**Reply by ahetawal-p** [reply]
Hmm not sure I follow, so does my input text looks like this ? (Doesnt work...)
```
<Radio.Group
      label="Select your favorite framework/library"
      description="This is anonymous"
      withAsterisk
    >
      <Radio value="react" label="React" />
      <Radio value="svelte" label="Svelte" />
      <Radio value="other" label="other"> <TextInput />
</Radio>
    </Radio.Group>
```

**Reply by ahetawal-p** [reply]
@rtivital If it helps this is what I am trying to achieve


<img width="242" alt="Screen Shot 2022-08-21 at 7 04 53 AM" src="https://user-images.githubusercontent.com/8919529/185794880-b5e0aca7-cda3-44e1-8fdf-c3888c1caab8.png">

The above code certainly doesnt work, any other ways to achieve this UI ?

**Reply by rtivital** [reply]
Wrap Radio and TextInput with Group
